editor/components/rich-text/index.native.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import RCTAztecView from 'react-native-aztec';
+import { parse, children } from '@wordpress/blocks';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, RawHTML, renderToString } from '@wordpress/element';
+import { withInstanceId, compose } from '@wordpress/compose';
+import deprecated from '@wordpress/deprecated';
+
+export class RichText extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onChange = this.onChange.bind( this );
+		this.onContentSizeChange = this.onContentSizeChange.bind( this );
+
+		this.lastContentSizeHeight = -1;
+		this.lastEventCount = 0;
+	}
+
+	/**
+	 * Handles any case where the content of the AztecRN instance has changed.
+	 */
+
+	onChange( event ) {
+		if ( !! this.currentTimer ) {
+			clearTimeout( this.currentTimer );
+		}
+		this.lastEventCount = event.nativeEvent.eventCount;
+		this.lastContent = event.nativeEvent.text;
+
+		this.currentTimer = setTimeout( function() {
+			// Create a React Tree from the new HTML
+			const newParaBlock = parse( '<!-- wp:paragraph --><p>' + this.lastContent + '</p><!-- /wp:paragraph -->' )[ 0 ];
```

Comment:
We should change this asap, since will only work when RichText is used in a para block.